conda/plugins/__init__.py

```diff
@@ -36,3 +36,23 @@ def conda_subcommands() -> Iterable[CondaSubcommand]:
 
     :return: An iterable of subcommand entries.
     """
+
+
+class CondaVirtualPackage(NamedTuple):
+    """
+    Conda subcommand entry.
+
+    :param name: Virtual package name (e.g., ``my_custom_os``).
+    :param version: Virtual package version (e.g., ``1.2.3``).
+    """
+    name: str
+    version: Optional[str]
+
+
+@_hookspec
+def conda_virtual_package_plugin() -> Iterable[CondaVirtualPackage]:
```

Comment:
For the subcommands we used `conda_subcommands` as the name, so we should stay consistent and not use the `_plugin` suffix but `conda_virtual_packages` (plural since it can `yield` multiple).